g-compiler.c

```diff
@@ -58,11 +59,11 @@ int main(int argc, char **argv)
 
     int token;
     while ((token = yylex(lexer))) {
-        if (token == -1) {
+        if (-1 == token) {
             break;
         }
 
-        char *text = yyget_text(lexer);
```

Comment:
This duplicates the string for further usage inside the parser. E.g. safe it as a value.